bastion/gui_manager.py

```
@@ -49,25 +49,191 @@ def _executable_exists(self, path):
+                            tokens = line.replace('\0', ' ').split()
+                            for tok in tokens:
+                                if tok.startswith('DISPLAY='):
+                                    display = tok.split('=', 1)[1]
+                                    if display:
+                                        env['DISPLAY'] = display
+                                        logger.info(f"Found DISPLAY from ps: {display}")
+                                        display_found = True
+                                        break
+                            if display_found:
+                                break
+                except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)